### ISSUE TITLE
Broken pipe consuming events doesn't close connection

### ISSUE THREAD

**Issue by blalor** [fsouza/go-dockerclient#202]
I've been working with progrium/registrator and running into cases where Registrator no longer handles events, even though I'm stopping/starting containers.  I've just upgraded Registrator to use the current code for `go-dockerclient` and ran into the problem again with docker 1.3.2.  In this case, I found the following in the docker daemon log:

```
[2f7f65a1] -job rm(log-ingester) = OK (0)
write unix @: broken pipe
[2f7f65a1] -job events() = ERR (1)
[error] server.go:1207 Handler for GET /events returned error: write unix @: broken pipe
[error] server.go:110 HTTP Error: statusCode=500 write unix @: broken pipe
2015/01/12 19:56:54 http: multiple response.WriteHeader calls
```

That corresponds to the time when Registrator became unresponsive.  This looks like it could be a Docker problem, but with those logs it's almost impossible to tell.  Anyway, it seems that the connection's been closed but go-dockerclient hasn't handled it.  Specifically, the [channel consuming the events](https://github.com/progrium/registrator/blob/6a1053c48e17c597f68cbf5c42d702136bdf5dd9/registrator.go#L116) should get closed.

Thoughts on how to debug this?

**Comment by amasad** [comment]
Running into the same problem, I think, I stop getting an events via the event listener API.

```
Handler for GET /events returned error: write unix @: broken pipe 
```

**Comment by blalor** [comment]
This may have been fixed in #206.  Do you have that commit in the version you're running from?

**Comment by amasad** [comment]
Apparently not. I'll update. Thanks
